Library/Homebrew/brew.sh

```diff
@@ -294,6 +289,21 @@ EOS
 }
 check-run-command-as-root
 
+check-prefix-is-not-tmpdir() {
+  [[ -z "${HOMEBREW_MACOS}" ]] && return
+
+  if [[ "${HOMEBREW_PREFIX}" = "${HOMEBREW_TEMP}"* ]]
+  then
+    odie <<EOS
+Your HOMEBREW_PREFIX is in the system temporary directory, which Homebrew
```

Comment:
`HOMEBREW_TEMP` isn't necessary the _system_ temporary directory. To expand on my previous comment (and avoid the autohiding): is the issue if:

- the prefix is in the system temporary directory
- the prefix is the in the HOMEBREW_TEMP temporary directory
- either

We should ensure that the wording makes that clear and the logic is as minimal as possible to the actual problems we've seen people experience.

Reply:
The issue is when the prefix is in the `HOMEBREW_TEMP` temporary directory. I'll correct the language here :slightly_smiling_face: